### PR TITLE
Challenge detail feature.

### PR DESCRIPTION
**Pull request by isht3**
`evalai challenge CHALLENGE`

### CODE REVIEW

**Comment by RishabhJain2018** [comment]
@isht3 I tried to review it but it is giving me error:
screen shot 2018-07-04 at 11 12 26 pm

**Comment by isht3** [comment]
@RishabhJain2018 This will be rectified after submissions PR is merged.

**Comment by isht3** [comment]
@RishabhJain2018 Updated.

**Comment by RishabhJain2018** [comment]
@isht3 Some conflicts!

**Comment by isht3** [comment]
@RishabhJain2018 Done!

**Comment by RishabhJain2018** [comment]
@isht3 I tried to test it on my local but the current content looks ugly. Here is the screenshot. Please add it in tabular format.
screen shot 2018-07-05 at 10 15 33 pm

**Comment by isht3** [comment]
@RishabhJain2018 Table format won't look good because there is only one challenge and most of the attributes are big text paras.

**Comment by isht3** [comment]
@RishabhJain2018 I've updated the PR with HTML parsing. Can you tell me whether it's okay now?

**Comment by RishabhJain2018** [comment]
@isht3 No, this is also not looking good, let's try table format once!
screen shot 2018-07-06 at 3 46 32 pm

**Comment by isht3** [comment]
This is how it looks. It may look good if we leave out start date, end date and created by.

**Comment by isht3** [comment]
@RishabhJain2018 Should I go by this?

**Comment by RishabhJain2018** [comment]
This looks organised than what we had earlier.
We can drop the `creator` field and also please try to increase the width of the table and paste the screenshot.

**Comment by isht3** [comment]
I dropped the ID and creator, since Id is not required here.

**Comment by isht3** [comment]
@RishabhJain2018 I've made the table changes you mentioned. Can you check whether if it's okay now?

File: evalai/utils/challenges.py [hunk 1]
```diff
@@ -194,6 +194,48 @@ def display_participated_or_hosted_challenges(is_host=False, is_participant=Fals
             echo("Sorry, no challenges found!")
 
 
+def pretty_print_challenge_details(challenge):
+    table = BeautifulTable(max_width=210)
+    attributes = ["description", "submission_guidelines", "evaluation_details", "terms_and_conditions"]
+    column_attributes = ["Title", "Start Date", "End Date", "Description", "Submission Guidelines",
+                         "Evaluation Details", "Terms and Conditions"]
+    table.column_headers = column_attributes
```
Comment (varunagrawal) [inline evalai/utils/challenges.py]: Why not directly set `table.column_headers` to the list? I don't see `column_attributes` being used elsewhere.

Comment (isht3) [reply evalai/utils/challenges.py]: Didn't notice that. Will change it!

Comment (isht3) [reply evalai/utils/challenges.py]: Done

File: evalai/utils/challenges.py [hunk 2]
```diff
@@ -194,6 +194,47 @@ def display_participated_or_hosted_challenges(is_host=False, is_participant=Fals
             echo("Sorry, no challenges found!")
 
 
+def pretty_print_challenge_details(challenge):
+    table = BeautifulTable(max_width=210)
```
Comment (RishabhJain2018) [inline evalai/utils/challenges.py]: Please make it `max_width=200`

File: evalai/utils/challenges.py [hunk 3]
```diff
@@ -194,6 +194,47 @@ def display_participated_or_hosted_challenges(is_host=False, is_participant=Fals
...
+    attributes = ["description", "submission_guidelines", "evaluation_details", "terms_and_conditions"]
+    table.column_headers = ["Title", "Start Date", "End Date", "Description", "Submission Guidelines",
```
Comment (RishabhJain2018) [inline evalai/utils/challenges.py]: I think we can remove `Title` from here as it gets displayed when a user lists all challenges. I am asking this as it is distorting the complete table.

**Pull request merged**